maxim/blocks/grid_gating.py

```diff
@@ -6,7 +6,7 @@
 from tensorflow.keras import backend as K
 from tensorflow.keras import layers
 
-from ..layers import BlockImages, SwapAxes, UnblockImages
+from ..layers import SwapAxes, TFBlockImagesByGrid, TFUnblockImages
```

Comment:
Why there's a separate layer for handling blocking by grids?

Reply:
BlockByGrid can be implemented as follows: 
```
def BlockByGrid(image, grid_size):
    block_size = (image_height // grid_size[0], image_width // grid_size[1])
    return BlockImage(image, block_size)
```

But, while implementing TFBlockImages I used [tf.split](https://github.com/sayakpaul/maxim-tf/pull/24/files/d55b45acb0ff8287445a1df09104fc7fb9238c67#diff-52bd614bb8d4ae5da8df6a1a882250282cc2c6dfef96e6c904f1a12e08c586a6R23) which expects a int literal as argument for [num_or_size_splits](https://www.tensorflow.org/api_docs/python/tf/split).

However, in cases where we only have the grid_size and the block_size has to be computed in the fly, it needs to be a tensor, and we can't use tf.split ins this case. That's why I also wrote BlockByGrid.